robot_framework/process.py

```diff
@@ -1,8 +1,237 @@
+                cpr_case_pair.phone_number = _get_phone_numbers(browser, cpr_case_pair.cpr)
+            else:
+                cpr_case_pair.phone_number = ["No phone number found."]
+        except NoSuchElementException:
+            cpr_case_pair.phone_number = ["An error occurred, please check manually."]
+
+
+def send_status_emails(email: EmailInput):
+    """Send an email to the requesting party and to the controller.
+
+    Args:
+        email: The email that has been processed.
+    """
+    with open(config.EMAIL_ATTACHMENT, "rb") as file:
+        smtp_util.send_email(
+            email.requester,
+            config.EMAIL_STATUS_SENDER,
+            config.EMAIL_COMPLETED_SUBJECT,
+            config.EMAIL_COMPLETED_BODY,
+            config.SMTP_SERVER,
+            config.SMTP_PORT,
+            False,
+            [smtp_util.EmailAttachment(file, config.EMAIL_ATTACHMENT)]
+        )
+
+
+def _get_phone_numbers(browser: webdriver.Chrome, cpr_in: str) -> list[str]:
+    """Search for a cpr number on an already open case and extract the phone numbers associated
+    with the person.
+
+    Args:
+        browser: The browser object.
+        cpr_in: The cpr number to look for.
+
+    Returns:
+        The persons phone and mobile numbers.
+    """
+    table = browser.find_element(By.ID, "ctl00_ContentPlaceHolder2_GridViewMovingPersons")
+    rows = table.find_elements(By.TAG_NAME, "tr")
+    cpr_in = cpr_in.replace("-", "")
+
+    # Remove header row
+    rows.pop(0)
+
+    for row in rows:
+        cpr_link = row.find_element(By.XPATH, "td[2]/a[2]")
+        cpr = cpr_link.text.replace("-", "")
+
+        if cpr == cpr_in:
+            cpr_link.click()
+            break
+
+    phone_number = browser.find_element(By.ID, "ctl00_ContentPlaceHolder2_ptFanePerson_stcPersonTab1_lblTlfnrTxt").text
+    mobile_number = browser.find_element(By.ID, "ctl00_ContentPlaceHolder2_ptFanePerson_stcPersonTab1_lblMobilTxt").text
+
+    numbers = []
+    if phone_number:
+        numbers.append(phone_number)
+    if mobile_number:
+        numbers.append(mobile_number)
+
+    return numbers
+
+
+def write_excel(cases: list[CprCase]) -> BytesIO:
+    """Write a list of task objects to an excel sheet.
+
+    Args:
+        tasks: The list of task objects to write.
+
+    Returns:
+        A BytesIO object containing the Excel sheet.
+    """
+    wb = Workbook()
+    sheet: Worksheet = wb.active
+    header = ["Sagsnr.", "CPR", "Navn", "Telefonnumre"]
+    sheet.append(header)
+
+    # Populate the sheet
+    for cpr_case in cases:
+        phone_numbers = convert_phone_number((cpr_case.phone_number))
+        row = [cpr_case.case, cpr_case.cpr, cpr_case.name, phone_numbers]
+        sheet.append(row)
+
+    # Styling
+    set_column_width(sheet)
+    for cell in sheet[1]:
+        cell.font = Font(bold=True)
+
+    # Save the file
+    file = BytesIO()
+    wb.save(file)
+    with open(config.EMAIL_ATTACHMENT, 'wb') as f:
```

Comment:
There's no reason to write the file to disk. Both smtp_util and openpyxl can work with files in memory